lib/matplotlib/cbook.py

```diff
@@ -658,6 +658,31 @@ def __setstate__(self, state):
         self.extend(state['seq'])
 
 
+class IgnoredKeywordWarning(UserWarning):
+    """
+    A class for issuing warnings about keyword arguments that will be ignored
+    by matplotlib
+    """
+    pass
+
+
+def kwarg_or_local(kwargs, kwarg_name, local_var=None):
+    """
+    Enforces the priority of a local variable over a potentially conflicting
+    argument in a kwargs dict, but falls back on the value of the keyword
+    argument if the local variable's value is None. Raises an
+    IgnoredKeywordWarning if the keyword argument is specified but ignored.
```

Comment:
Can you please add numpydoc style Paramaters and Returns sections to this docstring?